webpack/assets/javascripts/react_app/components/common/TypeAheadSelect/index.js

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { initialUpdate, updateSelected } from './TypeAheadSelectActions';
+import { selectOptions, selectSelected } from './TypeAheadSelectSelectors';
+import reducer from './TypeAheadSelectReducer';
+import TypeAheadSelect from './TypeAheadSelect';
+
+const ConnectedTypeAheadSelect = ({
+  id,
+  options,
+  selected,
+  defaultSelected,
+  allowNew,
+  multiple,
+}) => {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(initialUpdate(options, selected, id));
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Should we also return a cleanup unmount action here?

Reply:
Not sure how to make that determination. Do you think it's worth it? What's the disadvantage of not doing it?

Reply:
If you return in the end of the useEffect, it will be treated as componentWillUnmount.

> What's the disadvantage of not doing it?

If you had some data in the redux state, it may replace the actual data that was recieved from props for a new component.. it is just a precaution

Reply:
It would be a very good practice, cleaning the state when the component get unmounted.
Especially once/if we will move more pages to react-router so we will keep the global state when moving between pages.

Reply:
Understood. But why not add it later when it's needed? I'm typically not a fan of adding code because it will have value at some point in the future.

Also, I took a quick look and it doesn't seem we're doing this on other 'useEffect' cases. Since the components in store are unique by their ID , is this cleanup truly necessary?

Reply:
@laviro @sharvit  I think that 

* since we don't have any timeout intervals or API calls to clean up, and
* the `id` prop is used to differentiate between multiple TypeAheadSelect components,

using the cleanup feature of useEffect is not essential in this particular case.  It might even be a convenience to the user in the future, if they are able to return to this page and see what they were in the middle of doing.  I do agree in general that it's a good practice though.